src/architect/project/project.utils.ts

```diff
@@ -0,0 +1,290 @@
+        };
+      }
+    }
+    return selections;
+  }
+
+  static async prompt(choices: any[], message: string): Promise<any> {
+    inquirer.registerPrompt('autocomplete', require('inquirer-autocomplete-prompt'));
+    const answers: { selected: any } = await inquirer.prompt([
+      {
+        type: 'autocomplete',
+        name: 'selected',
+        message: message,
+        source: async (answers_so_far: any, input: string) => {
+          return choices.map((p) => ({ name: p.name, value: p }));
+        },
+      },
+    ]);
+    return answers.selected;
+  }
+
+  static async downloadRepo(url: string, dir_path: string): Promise<void> {
+    await download(url + `#${GITHUB_BRANCH}`, dir_path, (err: any) => {
+      if (err) {
+        throw new Error(`Failed to download repository ${url}`);
+      }
+    });
+  }
+
+  static async downloadGitHubRepos(selections: Dictionary<Dictionary<any>>, project_dir: string): Promise<void> {
+    // download any selection that has a homepage
+    for (const selection of Object.values(selections)) {
+      if (selection.architect_yml && selection.architect_yml.homepage) {
+        const url = selection.architect_yml.homepage.replace('https://github.com/', '');
+        await this.downloadRepo(url, project_dir + '/' + selection.name.toLowerCase());
+      }
+    }
+    await new Promise(resolve => setTimeout(resolve, 2000));
+  }
+
+  static async createArchitectYaml(selections: Dictionary<any>, project_dir: string): Promise<void> {
+    // get actual service name for depends_on
+    for (const selection of Object.values(selections)) {
+      if (selection.depends_on) {
+        const depends_on_proj = selections[selection.depends_on];
+        const services = depends_on_proj.architect_yml.services;
+        if (Object.keys(services).length === 1) {
+          selection.depends_on = Object.keys(services)[0];
+        } else {
+          const service_name = Object.keys(services).find((name: any) => !services[name].image);
+          selection.depends_on = service_name;
+        }
+      }
+    }
+
+    const combined_yml: Dictionary<any> = {};
+    combined_yml['name'] = project_dir.replace('_', '-');
+    const ignored = ['name', 'description', 'homepage', 'keywords'];
+    for (const selection of Object.values(selections)) {
+      for (const [key, val] of Object.entries(selection.architect_yml as ComponentSpec)) {
+        if (ignored.includes(key)) {
+          continue;
+        }
+
+        if (key !== 'services') {
+          combined_yml[key] = combined_yml[key] ? { ...combined_yml[key], ...val } : val;
+          continue;
+        }
+
+        const yml_services = val as Dictionary<ServiceSpec>;
+        const service_keys = Object.keys(yml_services);
+        if (service_keys.length === 1) {
+          const service_key = service_keys[0];
+          const service = yml_services[service_key];
+          if (service.build) {
```

Comment:
Why do we need to do this?

Reply:
It is for handling where to put the `depends_on` but since we will be updating how we handle `depends_on`, I will remove that.